scripts/scil_resample_tractogram.py

```diff
@@ -0,0 +1,125 @@
+Script to resample a tractogram to a set number of streamlines.
+Can be useful to build training sets for machine learning algorithms, to
+upsample under-represented bundles or downsample over-represented bundles.
+
+Works by either selecting a subset of streamlines or by generating new
+streamlines by adding gaussian noise to existing ones.
+
+Includes smoothing to compensate for the noisiness of new streamlines
+generated by the process.
+
+Example usage:
+$ scil_resample_tractogram.py input.trk 1000 output.trk \
+--point_wise_std 0.5 --spline 5 10 --keep_invalid_streamlines
+$ scil_visualize_bundles.py output.trk --local_coloring --width=0.1
+"""
+
+import argparse
+import logging
+
+from dipy.io.stateful_tractogram import StatefulTractogram
+from dipy.io.streamline import save_tractogram
+from dipy.tracking.streamlinespeed import compress_streamlines
+
+from scilpy.io.streamlines import load_tractogram_with_reference
+from scilpy.io.utils import (add_overwrite_arg, add_reference_arg,
+                             add_verbose_arg,
+                             assert_inputs_exist,
+                             assert_outputs_exist)
+
+from scilpy.tracking.tools import get_subset_streamlines
+from scilpy.utils.streamlines import upsample_tractogram
+
+
+def _build_arg_parser():
+    p = argparse.ArgumentParser(formatter_class=argparse.RawTextHelpFormatter,
+                                description=__doc__)
+
+    p.add_argument('in_tractogram',
+                   help='Input tractography file.')
+    p.add_argument('nb_streamlines', type=int,
+                   help='Number of streamlines to resample the tractogram to.')
+    p.add_argument('out_tractogram',
+                   help='Output tractography file.')
+    std_group = p.add_mutually_exclusive_group()
+    std_group.add_argument('--point_wise_std', type=float,
+                           help='Noise to add to existing streamlines\'' +
+                                'points to generate new ones.')
+    std_group.add_argument('--streamline_wise_std', type=float,
+                           help='Noise to add to existing whole' +
+                                ' streamlines to generate new ones.')
+
+    sub_p = p.add_mutually_exclusive_group()
+    sub_p.add_argument('--gaussian', metavar='SIGMA', type=int,
+                       help='Sigma for smoothing. Use the value of surronding'
+                            ' X,Y,Z points on the streamline to blur the'
+                            ' streamlines.\nA good sigma choice would be '
+                            'around 5.')
+    sub_p.add_argument('--spline', nargs=2, metavar=('SIGMA', 'NB_CTRL_POINT'),
+                       type=int,
+                       help='Sigma for smoothing. Model each streamline as a '
+                            'spline.\nA good sigma choice would be around 5 '
+                            'and control point around 10.')
+
+    p.add_argument('-e', dest='error_rate', type=float, default=0.1,
```

Comment:
-e or --e  ?

Reply:
Please remove the compressing.

Reply:
Donesies in df4d41a44f3f2bfd5fcf3c4899142cfeccbc5675